src/onelogin/saml2/idp_metadata_parser.py

```diff
@@ -108,9 +126,12 @@ def parse(idp_metadata):
 
                     if idp_entity_id is not None:
                         data['idp']['entityId'] = idp_entity_id
+
                     if idp_sso_url is not None:
                         data['idp']['singleSignOnService'] = {}
                         data['idp']['singleSignOnService']['url'] = idp_sso_url
+                        data['idp']['singleSignOnService']['binding'] = preferred_sso_binding
```

Comment:
Well, this comment depends on the fact that you fix and retrieve the 1st element, you will need to take care of that, I only wanted to notify that fact to you, to keep that in mind.